[meta-oe] Introduce recipe for sdbus-c++ library and its tools

sangelovic opened the pull request:
sdbus-c++ is a high-level C++ D-Bus library written on top of systemd D-Bus implementation originally as a modern, robust replacement for dbus-c++.

There are two main recipes and one helper recipe:
* Recipe for sdbus-c++ itself. It detects whether systemd feature is turned on. If so, it depends on systemd and links against libsystemd. Otherwise it builds libsystemd as part of building sdbus-c++. (Building of libsystemd for sdbus-c++ is performed by helper `sdbus-c++-libsystemd` recipe.) The recipe also supports ptest.
* Recipe for sdbus-c++ native tools, namely sdbus-c++ code generator to generate C++ adaptor and proxy binding classes.

The recipes have been tested on yocto master.

kraj commented on meta-oe/recipes-core/sdbus-c++/sdbus-c++-tools-native_0.7.1.bb:
```diff
@@ -0,0 +1,17 @@
+
+inherit cmake native
+
+DEPENDS_${PN} = "expat"
```
this should just be `DEPENDS += "expat"`

sangelovic replied:
Fixed

kraj commented on meta-oe/recipes-core/sdbus-c++/sdbus-c++-tools-native_0.7.1.bb:
```diff
@@ -0,0 +1,17 @@
+RDEPENDS_${PN} = "expat"
+
+BRANCH = "master"
+TAG = "v${PV}"
```
Please use hardcoded git SHA instead of tags, tags are floating.

sangelovic replied:
Done

kraj commented on meta-oe/recipes-core/sdbus-c++/sdbus-c++_0.7.1.bb:
```diff
@@ -0,0 +1,45 @@
+EXTRA_OECMAKE = "-DBUILD_CODE_GEN=ON \
+                 -DBUILD_DOC=ON \
+                 -DBUILD_DOXYGEN_DOC=OFF \
+                 ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', '-DBUILD_LIBSYSTEMD=OFF', '-DBUILD_LIBSYSTEMD=ON -DLIBSYSTEMD_VERSION=242', d)} \
```
hardcoding systemd version is going to be a maintenance pain since it has to be updated whenever systemd in OE-core is updated. Is there any other way to get version ? maybe something like
```pkg-config --modversion systemd```

sangelovic replied:
This version has nothing to do with systemd version built by separate recipe(s) in OE-core. It refers to version of just a small part of external libsystemd project that is built into sdbus-c++ library. But I removed that configuration option anyway, since sdbus-c++ has this libsystemd version as a default version in its makefiles.

sangelovic commented:
Hello, thanks for comments, and sorry for coming back to this after longer time. I was busy, but now is the time. I addressed your comments and at the same time bumped library version from v0.7.1 to v0.7.2 which contains an important fix. So the recipes are renamed. Please re-review... Thanks a lot!

kraj commented:
> Hello, thanks for comments, and sorry for coming back to this after longer time. I was busy, but now is the time. I addressed your comments and at the same time bumped library version from v0.7.1 to v0.7.2 which contains an important fix. So the recipes are renamed. Please re-review... Thanks a lot!

thanks for reworking on it. Can you squash all the commits into a single commit and upload again please

sangelovic commented:
> thanks for reworking on it. Can you squash all the commits into a single commit and upload again please

Done.

kraj commented on meta-oe/recipes-core/sdbus-c++/sdbus-c++_0.7.2.bb:
```diff
@@ -0,0 +1,46 @@
+inherit cmake pkgconfig systemd ptest python3native
+
+DEPENDS += "expat ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'systemd', 'meson-native ninja-native git-native gperf-native gettext-native util-linux glibc libcap', d)}"
+RDEPENDS_${PN} = "${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'libsystemd', 'glibc libcap', d)}"
```
I think hardcoding dependency on 'glibc' is not right. Did you mean something else here ?

sangelovic replied:
In this case sdbus-c++ build-depends and runtime-depends on librt library, which is AFAIK part of glibc. Do you have different suggestion in this regard?

kraj replied:
Librt is provided by c library and is not Glibc specific secondly dependency on Glibc are automatically added and are not required to be explicitly added

sangelovic replied:
So can I just remove glibc from `DEPENDS` and `RDEPENDS`?

sangelovic replied:
librt is provided by c library, but isn't glibc an implementation of c library for Linuxes?

kraj replied:
glibc is one of many, there is musl, there uclibc and maybe more. in OE we actively support musl and glibc, so hardcoding a libc dependency is not preferred unless its absolutely required then also we use COMPATIBLE_HOST variable to control the recipes for targets

sangelovic replied:
OK, glibc was removed.

kraj commented on meta-oe/recipes-core/sdbus-c++/sdbus-c++_0.7.2.bb:
```diff
@@ -0,0 +1,46 @@
+
+inherit cmake pkgconfig systemd ptest python3native
+
+DEPENDS += "expat ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'systemd', 'meson-native ninja-native git-native gperf-native gettext-native util-linux glibc libcap', d)}"
```
I think hardcoding dependency on 'glibc' is not right. Did you mean something else here ?
and I dont understand why these dependencies are conditional on systemd. They should be added
in general.

sangelovic replied:
Things are conditional because of flexibility. sdbus-c++and has two modes of building. One is when entire systemd ecosystem is available in the system. Then `libsystemd` shared library, the library with lower level D-Bus C implementation which sdbus-c++ depends on, is available, and will be used. But sdbus-c++ provides flexibility to be used on non-systemd systems (where there is no `libsystemd` library). In that case, sdbus-c++ build builds just the small needed part of entire systemd and incorporates it as part of sdbus-c++ lib, making sdbus-c++ independent of systemd. Hence this conditional logic in the recipe which checks whether systemd feature is on or off. The user of yocto doesn't care, they build sdbus-c++ into their OS and it works perfectly whether or not systemd ecosystem is present.

sangelovic replied:
Regarding glibc, see previous comment.

sangelovic replied:
When sdbus-c++ needs to build that sd-bus part of systemd, which is in libsystemd, it needs all those additional deps, like `meson-native ninja-native libpcap etc.`

kraj replied:
Okay and to be clear these dependencies are not needed when building without systemd

sangelovic replied:
> Okay and to be clear these dependencies are not needed when building without systemd

Yes, when building sdbus-c++ in a system where systemd feature is turned on (systemd is present), these dependencies are not selected, instead simply `systemd` dependency is selected.

kraj commented on meta-oe/recipes-core/sdbus-c++/sdbus-c++_0.7.2.bb:
```diff
@@ -0,0 +1,46 @@
+SECTION = "libs"
+
+LICENSE = "LGPLv2.1"
+LIC_FILES_CHKSUM = "file://${WORKDIR}/sdbus-cpp-${PV}/COPYING;md5=1803fa9c2c3ce8cb06b4861d75310742"
```
You dont need `${WORKDIR}/sdbus-cpp-${PV}/` here

sangelovic replied:
Fixed.

kraj commented on meta-oe/recipes-core/sdbus-c++/sdbus-c++-tools-native_0.7.2.bb:
```diff
@@ -0,0 +1,15 @@
+inherit cmake native
+
+DEPENDS += "expat"
+RDEPENDS_${PN} = "expat"
```
we only need this rdep it if tools are doing dlopen on libexpat, otherwise, shlib tool will automatically pick it up

sangelovic replied:
Fixed.

shr-project commented on meta-oe/recipes-core/sdbus-c++/sdbus-c++_0.7.2.bb:
```diff
@@ -0,0 +1,46 @@
+EXTRA_OECMAKE = "-DBUILD_CODE_GEN=ON \
+                 -DBUILD_DOC=ON \
+                 -DBUILD_DOXYGEN_DOC=OFF \
+                 ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', '-DBUILD_LIBSYSTEMD=OFF', '-DBUILD_LIBSYSTEMD=ON', d)} \
```
Use PACKAGECONFIG for this and move all extra build-time and runtime dependencies for libsystemd to PACKAGECONFIG.

sangelovic replied:
Done. Can you please review whether I've done that properly?

shr-project commented on meta-oe/recipes-core/sdbus-c++/sdbus-c++_0.7.2.bb:
```diff
@@ -0,0 +1,46 @@
+
+# Allow CMake to use git on the host machine
+OECMAKE_FIND_ROOT_PATH_MODE_PROGRAM = "BOTH"
+# Ninja is the default CMake generator in Yocto now, so let's set Unix Makefiles explicitly
```
Please say why, are the generated Makefiles incompatible with Ninja?

sangelovic replied:
When building the recipe with Ninja, I get: `ninja: error: 'libsystemd-v242/src/LibsystemdBuildProject-build/libsystemd.a', needed by 'libsdbus-c++.so.0.7.2', missing and no known rule to make it`.

sdbus-c++ uses ExternalProject feature of CMake and builds libsystemd using that. This is where it doesn't work with Ninja out of the box.

sangelovic commented:
I addressed all your comments, please check... Is it OK now?

kraj commented:
please squash all into single commit and push again

sangelovic commented:
> 
> 
> please squash all into single commit and push again

Done.

kraj commented:
here is new error

https://errors.yoctoproject.org/Errors/Details/253801/

sangelovic commented:
What do logs
at TOPDIR/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-stamp/LibsystemdBuildProject-configure-*.log
say?

On Tue, Jul 23, 2019, 16:28 Khem Raj <notifications@github.com> wrote:

> here is new error
>
> https://errors.yoctoproject.org/Errors/Details/253801/
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openembedded/meta-openembedded/pull/137?email_source=notifications&email_token=ABM3OUWZJIDRMDIBGNH6EUDQA4IPJA5CNFSM4HXHJPSKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2TJPWY#issuecomment-514234331>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABM3OUTKTZOMQPOLOVC6NR3QA4IPJANCNFSM4HXHJPSA>
> .
>

kraj commented:
CMake Error at /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-stamp/LibsystemdBuildProject-configure--impl.cmake:19 (message):
  Command failed (1):

   '/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot-native/usr/bin/meson' '--buildtype=plain' '-Dstatic-libsystemd=pic' '/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject' '/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build'

kraj commented:
`
The Meson build system
Version: 0.50.1
Source dir: /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject
Build dir: /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build
Build type: cross build
Project name: systemd
Project version: 242
Appending CFLAGS from environment: ' -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot=                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot-native=   -mcpu=cortex-a53+crc --rtlib=compiler-rt --stdlib=libc++ -mlittle-endian -Wno-error=unused-command-line-argument -Qunused-arguments -fstack-protector-strong  -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot'
Appending LDFLAGS from environment: '-Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fstack-protector-strong -Wl,-z,relro,-z,now'
Appending CFLAGS from environment: ' -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot=                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot-native=   -mcpu=cortex-a53+crc --rtlib=compiler-rt --stdlib=libc++ -mlittle-endian -Wno-error=unused-command-line-argument -Qunused-arguments -fstack-protector-strong  -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot'
Appending LDFLAGS from environment: '-Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fstack-protector-strong -Wl,-z,relro,-z,now'

meson.build:3:0: ERROR: Could not invoke sanity test executable: [Errno 8] Exec format error: '/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build/meson-private/sanitycheckc.exe'.

A full log can be found at /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build/meson-logs/meson-log.txt
`

sangelovic commented:
Thanks a lot. Would you please also paste the content
of /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build/meson-logs/meson-log.txt
here? That should be enough for me to start investigating the issue.

On Tue, Jul 23, 2019, 17:55 Khem Raj <notifications@github.com> wrote:

> `
> The Meson build system
> Version: 0.50.1
> Source dir:
> /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject
> Build dir:
> /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build
> Build type: cross build
> Project name: systemd
> Project version: 242
> Appending CFLAGS from environment: ' -O2 -pipe -g
> -feliminate-unused-debug-types
> -fmacro-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0
> -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0
> -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot=
> -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot-native=
> -mcpu=cortex-a53+crc --rtlib=compiler-rt --stdlib=libc++ -mlittle-endian
> -Wno-error=unused-command-line-argument -Qunused-arguments
> -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security
> -Werror=format-security
> --sysroot=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot'
> Appending LDFLAGS from environment: '-Wl,-O1 -Wl,--hash-style=gnu
> -Wl,--as-needed -fstack-protector-strong -Wl,-z,relro,-z,now'
> Appending CFLAGS from environment: ' -O2 -pipe -g
> -feliminate-unused-debug-types
> -fmacro-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0
> -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0
> -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot=
> -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot-native=
> -mcpu=cortex-a53+crc --rtlib=compiler-rt --stdlib=libc++ -mlittle-endian
> -Wno-error=unused-command-line-argument -Qunused-arguments
> -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security
> -Werror=format-security
> --sysroot=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot'
> Appending LDFLAGS from environment: '-Wl,-O1 -Wl,--hash-style=gnu
> -Wl,--as-needed -fstack-protector-strong -Wl,-z,relro,-z,now'
>
> meson.build:3:0: ERROR: Could not invoke sanity test executable: [Errno 8]
> Exec format error:
> '/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build/meson-private/sanitycheckc.exe'.
>
> A full log can be found at
> /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build/meson-logs/meson-log.txt
> `
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openembedded/meta-openembedded/pull/137?email_source=notifications&email_token=ABM3OUSI3FM7JIYHVMNC5GLQA4SWDA5CNFSM4HXHJPSKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2TSWEI#issuecomment-514272017>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABM3OURZYHXL2UPTPOT35NLQA4SWDANCNFSM4HXHJPSA>
> .
>

kraj commented:
```
Build started at 2019-07-23T06:32:58.982388
Main binary: /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot-native/usr/bin/nativepython3
Python system: Linux
The Meson build system
Version: 0.50.1
Source dir: /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject
Build dir: /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build
Build type: cross build
Project name: systemd
Project version: 242
Appending CFLAGS from environment: ' -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot=                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot-native=   -mcpu=cortex-a53+crc --rtlib=compiler-rt --stdlib=libc++ -mlittle-endian -Wno-error=unused-command-line-argument -Qunused-arguments -fstack-protector-strong  -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot'
Appending LDFLAGS from environment: '-Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fstack-protector-strong -Wl,-z,relro,-z,now'
Appending CFLAGS from environment: ' -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0=/usr/src/debug/sdbus-c++/0.7.2-r0                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot=                      -fdebug-prefix-map=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot-native=   -mcpu=cortex-a53+crc --rtlib=compiler-rt --stdlib=libc++ -mlittle-endian -Wno-error=unused-command-line-argument -Qunused-arguments -fstack-protector-strong  -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot'
Appending LDFLAGS from environment: '-Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fstack-protector-strong -Wl,-z,relro,-z,now'
Sanity testing C compiler: aarch64-yoe-linux-musl-clang -mcpu=cortex-a53+crc --rtlib=compiler-rt --stdlib=libc++ -mlittle-endian -Wno-error=unused-command-line-argument -Qunused-arguments -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot
Is cross compiler: False.
Sanity check compiler command line: aarch64-yoe-linux-musl-clang -mcpu=cortex-a53+crc --rtlib=compiler-rt --stdlib=libc++ -mlittle-endian -Wno-error=unused-command-line-argument -Qunused-arguments -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/recipe-sysroot /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build/meson-private/sanitycheckc.c -o /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build/meson-private/sanitycheckc.exe
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build/meson-private/sanitycheckc.exe

meson.build:3:0: ERROR: Could not invoke sanity test executable: [Errno 8] Exec format error: '/mnt/a/yoe/build/tmp/work/aarch64-yoe-linux-musl/sdbus-c++/0.7.2-r0/build/libsystemd-v242/src/LibsystemdBuildProject-build/meson-private/sanitycheckc.exe'.

```

kraj commented:
do you plan to submit a new version with fixes

sangelovic commented:
Yes, I do. (Just didn't have much time for it at this recent summer time...)

On Wed, Aug 14, 2019, 07:02 Khem Raj <notifications@github.com> wrote:

> do you plan to submit a new version with fixes
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openembedded/meta-openembedded/pull/137?email_source=notifications&email_token=ABM3OURYITOA2NSOEZAREDLQEOGVHA5CNFSM4HXHJPSKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4HWHKI#issuecomment-521102249>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABM3OUVN4JOGD7PE76QOLB3QEOGVHANCNFSM4HXHJPSA>
> .
>

sangelovic commented:
Hi Khem, I finally got time to address sdbus-c++ recipe review comments and ARM build error. I have significantly refactored the recipe. Now systemd library is built by a specific auxiliary recipe of its own - `sdbus-c++-libsystemd` (version of libsystemd is 242, because that's current stable version for sdbus-c++, but I'll check 243 very soon). sdbus-c++ will just depend on it, in case systemd feature is off. I've also addressed review comments of Stefan Herbrechtsmeier.

Hopefully we are getting to final accepted state of sdbus-c++ recipes.

Please could you re-review?

kraj commented:
https://errors.yoctoproject.org/Errors/Details/273663/

sangelovic commented:
Strange. I need to reproduce this locally, I have no experience with
raspberry target. How can I reproduce this? What settings shall I apply in
my local yocto build? Thank you...

On Thu, Oct 17, 2019, 05:26 Khem Raj <notifications@github.com> wrote:

> https://errors.yoctoproject.org/Errors/Details/273663/
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openembedded/meta-openembedded/pull/137?email_source=notifications&email_token=ABM3OUX3YEBD3PGZZXDO2PTQO7LN3A5CNFSM4HXHJPSKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBOT5ZQ#issuecomment-542981862>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABM3OUQT64GYP7H46U2PHELQO7LN3ANCNFSM4HXHJPSA>
> .
>

sangelovic commented:
I tested ARM and x86 platforms, these work fine...

Missing printf.h is a strange problem. Given that an official release of
systemd is being built...

On Thu, Oct 17, 2019, 05:26 Khem Raj <notifications@github.com> wrote:

> https://errors.yoctoproject.org/Errors/Details/273663/
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openembedded/meta-openembedded/pull/137?email_source=notifications&email_token=ABM3OUX3YEBD3PGZZXDO2PTQO7LN3A5CNFSM4HXHJPSKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBOT5ZQ#issuecomment-542981862>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABM3OUQT64GYP7H46U2PHELQO7LN3ANCNFSM4HXHJPSA>
> .
>

kraj commented:
this is mostly because its using `TCLIBC = "musl" ` so you might want to set that in `local.conf`

sangelovic commented:
I've provided patches for MUSL. Please re-review...

kraj commented:
@sangelovic Please squash all the patches into a single patch and send the pull

sangelovic commented:
Done.

kraj commented:
I have cherry-picked it for master-next, in future add your sign-offs to commits, and commit msg guidelines are here [1]

[1] https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines

kraj commented:
fails to build with clang/libc++/musl

http://errors.yoctoproject.org/Errors/Details/275653/

kraj commented:
Fetch errors

http://errors.yoctoproject.org/Errors/Details/275690/

sangelovic commented:
Hi @kraj . Thanks for having patience with me :) I fixed the clang issues. I've also updated the commit message, I hope it's along the guidelines now. Please could you re-test? I hope this could finally make it...

sangelovic commented:
Regarding the fetch errors, I have no idea what the issue was. Perhaps just that github was down? Because the project path and the commit hash is valid...

kraj commented:
merged with https://github.com/openembedded/meta-openembedded/commit/b2baaae2711d098c06423858662e29a8375fe6b8
